tier4/tier4_perception_dataset
# refactor(rosbag2_to_non_annotated_t4_converter): refactor _convert_image for test

## Description

Refactor self._convert_image() function in rosbag2_to_non_annotated_t4_converter so that the LiDAR-camera synchronization algorithm can be tested with pytest. This PR is expected to be followed by another PR which enables a LiDAR-camera fusion w/ robustness to LiDAR message drops.


## How to review

- Run the test and confirm that it passes
- Run a LiDAR-camera rosbag conversion
- Run a camera-only rosbag conversion


## How to test

### test data

Any data that could be converted with the current latest tier4_perception_dataset is fine.


### test command



```bash
poetry run pytest tests/utils/test_misc.py
```

## Reference

[TIER IV INTERNAL LINK](https://star4.slack.com/archives/C03S84LDJGG/p1701929878859259)


## Notes for reviewer

None
